### PR TITLE
Java: add Java guide for Apache HttpClient based transport.

### LINKED ISSUE #3333: Java Server Performance docs: Point users to Apache transport
There's no reference to our Apache HTTP transport from the Java performance docs:

https://docs.sentry.io/platforms/java/configuration/transport/

We need to make sure to point users to this if they expect to instrument high throughput server apps with Java.

/cc @maciejwalkowiak

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
Fixes #3333

### CODE REVIEW

**Comment by maciejwalkowiak** [comment]
cc @marandaneto @bruno-garcia

**Review by bruno-garcia — APPROVED** [review]
LGTM but we probably should merge once the dep bump goes out to central

File: src/includes/performance/configure-sample-rate/java.mdx [hunk 1]
```diff
@@ -35,3 +35,49 @@ Sentry.init { options ->
   }
 }
 ```
+
+## High-Throughput Transport
```
Comment (marandaneto) [inline src/includes/performance/configure-sample-rate/java.mdx:39]: why don't we extract this to an includes file? the same content is within 3 files.

Also, I'm not entirely sure this lib is compatible with Android, so it should not be visible for Android, and right now it is

**Comment by marandaneto** [comment]
> LGTM but we probably should merge once the dep bump goes out to central

also, let's address my comment about Android at least.

**Comment by marandaneto** [comment]
moving to Draft till cutting a release

File: src/includes/performance/configure-sample-rate/java.mdx (hunk 1 above)
Comment (maciejwalkowiak) [reply src/includes/performance/configure-sample-rate/java.mdx:39]: In each of the guides this section is slightly different as there is a different way to setup transport in plain Java, Spring and Spring Boot.

Comment (maciejwalkowiak) [reply src/includes/performance/configure-sample-rate/java.mdx:39]: Android has a separate file for configuring sample rate, so this section does not appear there. Or you meant something else?

Comment (marandaneto) [reply src/includes/performance/configure-sample-rate/java.mdx:39]: I (almost) swear I've seen it on https://sentry-docs-git-fork-maciejwalkowiak-gh-java-3333.sentry.dev/platforms/android/performance/
but you're right, my bad, it's Friday!

**Review by marandaneto — APPROVED** [review]
LGTM

**Pull request closed without merging**